File: app/src/main/java/com/gwell/iotvideodemo/WelcomeActivity.java
package com.gwell.iotvideodemo;

import android.content.Intent;
import android.os.Bundle;

import com.gwell.iotvideo.accountmgr.AccountMgr;
import com.gwell.iotvideodemo.accountmgr.login.LoginActivity;
import com.gwell.iotvideodemo.base.BaseActivity;

public class WelcomeActivity extends BaseActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_welcome);

        boolean isLogin = AccountMgr.getInstance().isLogin(this);
        if (!isLogin) {
            startActivity(new Intent(this, LoginActivity.class));
        } else {
            startActivity(new Intent(this, MainActivity.class));
        }
        finish();
    }

    @Override
    public void onBackPressed() {

    }
}

File: app/src/main/java/com/gwell/iotvideodemo/WelcomeActivity.kt
package com.gwell.iotvideodemo

import android.content.Intent
import android.os.Bundle
import com.gwell.iotvideo.utils.UrlHelper

import com.gwell.iotvideodemo.accountmgr.AccountSPUtils
import com.gwell.iotvideodemo.accountmgr.login.LoginActivity
import com.gwell.iotvideodemo.base.BaseActivity
import com.gwell.iotvideodemo.utils.AppSPUtils

class WelcomeActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_welcome)
        val needChangeService = AppSPUtils.getInstance().getBoolean(this, AppSPUtils.NEED_SWITCH_SERVER_TYPE, false)
        when {
            needChangeService -> {
                AppSPUtils.getInstance().putBoolean(this, AppSPUtils.NEED_SWITCH_SERVER_TYPE, false)
                AccountSPUtils.getInstance().putInteger(this, AccountSPUtils.VALIDITY_TIMESTAMP, 0)
                startActivity(Intent(this, LoginActivity::class.java))
            }
            AccountSPUtils.getInstance().isLogin(this) -> startActivity(Intent(this, MainActivity::class.java))
            else -> startActivity(Intent(this, LoginActivity::class.java))
        }
        finish()
    }

    override fun onBackPressed() {

    }
}
